kv/txn.go

```diff
@@ -38,35 +38,34 @@ func RunInNewTxn(store Storage, retryable bool, f func(txn Transaction) error) e
 			return errors.Trace(err)
 		}
 
+		// originalTxnTS is used to trace the original transaction when the function is retryable.
 		if i == 0 {
 			originalTxnTS = txn.StartTS()
 		}
 
 		err = f(txn)
-		if retryable && IsRetryableError(err) {
-			log.Warnf("[kv] Retry txn %v original txn %v err %v", txn, originalTxnTS, err)
-			err1 := txn.Rollback()
```

Comment:
why `txn.Rollback()` is removed ?

Reply:
it's not removed, it's merged into `if err != nil` at line 47